src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1713,8 +1713,10 @@ private bool SkipComment()
         private bool SkipSingleLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
         {
             idx = localBuffer.IndexOfAny(JsonConstants.LineFeed, JsonConstants.CarriageReturn);
+            int toConsume = 0;
```

Comment:
I could be wrong, but it looks like `toConsume` is redundant. Why not continue to use/update `idx`?

Reply:
`idx` has to point right before the line separator which could be `\r` or `\n` (1 char each) or `\r\n` (2 char sequence) so that the caller can extract just the comment text. At line [1756](https://github.com/dotnet/corefx/pull/35934/files/4c9d01e68525cfaa487e4016e217af672bc54ab9#diff-dbc577928a29b3154f5b195f56722391R1756) either 1 or 2 characters have to be _consumed_ depending on the line separator that was encountered. _toConsume_ captures that. At that code point this information would be lost if it were not tracked. Updating `idx` would not help as it would have to be reverted before function returned or somehow the caller would have to know if 1 or 2 characters are to be left out.
Thus, having this variable helps.

Reply:
`toConsume` is set to `idx` or `idx + 1` depending on whether the line separator is single character (i.e. `\r` or `\n`) i.e. at line [1719](https://github.com/dotnet/corefx/pull/35934/files?file-filters%5B%5D=#diff-dbc577928a29b3154f5b195f56722391R1719) and [1740](https://github.com/dotnet/corefx/pull/35934/commits/54e34296f86a75cc05298f840cc2f7a2e1e77bfd#diff-dbc577928a29b3154f5b195f56722391R1740) or a double character (`\r\n`) i.e. at  [1732](https://github.com/dotnet/corefx/pull/35934/commits/54e34296f86a75cc05298f840cc2f7a2e1e77bfd#diff-dbc577928a29b3154f5b195f56722391R1732), respectively.

So `toConsume` is not same as `idx` always. Please note that `idx` points to first line separator character after end of comment data, which the caller is interested in to extract just the comment data excluding comment characters and line separator whereas `toConsume` is number of line separator characters to be added to `_consumed`.